app/utilities/types.py

```diff
@@ -1,11 +1,22 @@
-from typing import NamedTuple, TypeAlias
+from typing import TYPE_CHECKING, NamedTuple, TypeAlias, Union
 
-from app.questionnaire.location import Location
-from app.questionnaire.relationship_location import RelationshipLocation
+if TYPE_CHECKING:
+    from app.questionnaire.location import Location
+    from app.questionnaire.relationship_location import RelationshipLocation
 
-LocationType: TypeAlias = Location | RelationshipLocation
+LocationType: TypeAlias = Union["Location", "RelationshipLocation"]
 
 
 class SectionKey(NamedTuple):
+    section_id: str
+    list_item_id: str | None = None
+
+
+class DependentSection(NamedTuple):
     section_id: str
     list_item_id: str | None
+    is_complete: bool | None
```

Comment:
These could be defaulted which would be result in around ~10 calls not needing to pass these. Left it out to minimise changes. Happy to address if reviewers are happy.

Reply:
Happy for a them to be defaulted if other devs are 👍

Reply:
Agreed, also prefer that 👍 
& thanks for explaining - it seemed odd its never True but I can't see a natural True/False replacement that makes sense so see why that is - maybe just worth a comment on the class to point that out.

Reply:
Addressed in [`f0e62fb` (#1196)](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1196/commits/f0e62fb583d789b08cb17aeeb0108f8c81351d6a) & [`8428314` (#1196)](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1196/commits/8428314c97ea119fb96dc962f34926f928f6c5bd)